modules/dnn/src/layers/max_unpooling_layer.cpp

```diff
@@ -71,9 +78,63 @@ class MaxUnpoolLayerImpl CV_FINAL : public MaxUnpoolLayer
         inputs_arr.getMatVector(inputs);
         outputs_arr.getMatVector(outputs);
 
-        CV_Assert(inputs.size() == 2);
-        Mat& input = inputs[0];
-        Mat& indices = inputs[1];
+        Mat input;
+        Mat indices;
+        if (inputs.size() == 2)
+        {
+            input = inputs[0];
+            indices = inputs[1];
+        }
+        else if (inputs.size() == 3)
+        {
+            input = inputs[2];
+
+            int kernel_area = poolKernel.height * poolKernel.width;
+            Mat maxpoolintp = inputs[0];
+            Mat maxpooloutp = inputs[1];
```

Comment:
May I ask you to just print these two matrices (with lower resolution input tensor)?
```cpp
std::cout << maxpoolintp.size << std::endl;
std::cout << maxpoolintp.reshape(1, 1) << std::endl;
std::cout << maxpooloutp.size << std::endl;
std::cout << maxpooloutpreshape(1, 1) << std::endl;
```

I think that one of them is constant and we can perform this layer using existing implementation.

Reply:
tensorflow's MaxUnpool need three inputs. The first parameter is the input of maxpool, the second paramter is the output of maxpool, the third paramter is the grad. 
for example:
_X_content = tf.placeholder(tf.float32, shape=(1, 256, 256, 3), name='X_content')
net = MaxPooling2D(pool_size=(2, 2), strides=(2, 2))(X_content)
net = MaxUnPooling2D(net.op, net)_
The first parameter is **net.op.input**(1, 256, 256, 3), the second paramter is **net.op.output**(1, 128, 128, 3) and the third paramter **net**(1, 128, 128, 3)

Reply:
> the second paramter is the output of maxpool

I think that this is not an output values but indices.

Reply:
we need use the first and second parameter to calc indices. This code works fine on my side.